### PR TITLE
feat: Multiple payment gateways

### PR DESCRIPTION
**Pull request by meaziz**
In shopify Integration 
Add support to multiple Payment Gateways that allows booking for payment for shopify orders across multiple accounts

This can be used for Cash on delivery, Credit Card payment etc...

### CODE REVIEW

**Comment by meaziz** [comment]
@ankush  Can you provide your feedback to this?

File: ecommerce_integrations/shopify/invoice.py [hunk 1]
```diff
@@ -63,13 +63,23 @@ def set_cost_center(items, cost_center):
...
+				_bank_account = gateway.get("gateway_account")
+
+	payment_entry = get_payment_entry(doc.doctype, doc.name, bank_account=_bank_account)
 	payment_entry.flags.ignore_mandatory = True
 	payment_entry.reference_no = doc.name
 	payment_entry.posting_date = posting_date or nowdate()
 	payment_entry.reference_date = posting_date or nowdate()
 	payment_entry.insert(ignore_permissions=True)
 	payment_entry.submit()
+	frappe.db.commit()
```
Comment (ankush) [inline ecommerce_integrations/shopify/invoice.py:85]: This shouldn't be added 

https://frappeframework.com/docs/v14/user/en/api/database#database-transaction-model

File: ecommerce_integrations/shopify/invoice.py [hunk 2]
```diff
@@ -63,13 +63,23 @@ def set_cost_center(items, cost_center):
 		item.cost_center = cost_center
 
 
-def make_payament_entry_against_sales_invoice(doc, setting, posting_date=None):
+def make_payament_entry_against_sales_invoice(doc, setting, shopify_order, posting_date=None):
 	from erpnext.accounts.doctype.payment_entry.payment_entry import get_payment_entry
+	_bank_account = setting.get('cash_bank_account')
 
-	payment_entry = get_payment_entry(doc.doctype, doc.name, bank_account=setting.cash_bank_account)
+	if len(shopify_order.get('payment_gateway_names')):
```
Comment (ankush) [inline ecommerce_integrations/shopify/invoice.py:70]:
```suggestion
	if len(shopify_order and shopify_order.get('payment_gateway_names') or []):
```

Handle case when it's not sent.

File: ecommerce_integrations/shopify/invoice.py [hunk 3]
```diff
@@ -63,13 +63,23 @@ def set_cost_center(items, cost_center):
...
+		doc_gatewaty = shopify_order.get('payment_gateway_names')[0]
+
+	if len(setting.get("payment_gateways")):
+		for gateway in setting.get("payment_gateways"):
+			if doc_gatewaty == gateway.get("gateway_name"):
```
Comment (ankush) [inline ecommerce_integrations/shopify/invoice.py:75]: `doc_gateway` won't be set if order didn't have any. You'll get `'UnboundVariableError'`.

File: ecommerce_integrations/ecommerce_integrations/doctype/payment_gateways/payment_gateways.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+   "options": "Account"
+  }
+ ],
+ "index_web_pages_for_search": 1,
+ "istable": 1,
+ "links": [],
+ "modified": "2023-02-13 11:37:44.390686",
+ "modified_by": "Administrator",
+ "module": "Ecommerce Integrations",
+ "name": "Payment Gateways",
```
Comment (ankush) [inline ecommerce_integrations/ecommerce_integrations/doctype/payment_gateways/payment_gateways.json:33]: Let's rename this to something that won't crash with others. `Shopify Payment Gateways`?

**Review by ankush — CHANGES_REQUESTED** [review]

**Comment by ankush** [comment]
reopen when ready

**Pull request closed without merging**